spec/features/staff/beis_users_can_edit_a_submission_spec.rb

```diff
@@ -52,6 +52,34 @@
       expect(page).to have_content I18n.t("activerecord.errors.models.submission.attributes.deadline.not_in_past")
     end
 
+    scenario "setting a Submission's deadline changes its state to 'active'" do
+      PublicActivity.with_tracking do
+        user = create(:beis_user)
+        authenticate!(user: user)
+
+        visit organisation_path(user.organisation)
+
+        within "##{submission.id}" do
+          expect(page).to have_content I18n.t("label.submission.state.inactive")
+          click_on I18n.t("default.link.edit")
+        end
+
+        fill_in "submission[deadline(3i)]", with: "31"
+        fill_in "submission[deadline(2i)]", with: "1"
+        fill_in "submission[deadline(1i)]", with: "2021"
+
+        click_on I18n.t("default.button.submit")
+
+        expect(page).to have_content I18n.t("action.submission.update.success")
+        within "##{submission.id}" do
+          expect(page).to have_content I18n.t("label.submission.state.active")
+        end
+        auditable_event = PublicActivity::Activity.find_by(trackable_id: submission.id)
```

Comment:
Same thing here, I know it is a drag but I feel like the seperation of the behaviour warrants a seperate spec?